### DISCUSSION TITLE
Open IFC file

### DISCUSSION THREAD

**Discussion by moaciroj** [GeometryGym/GeometryGymIFC#50] [General]
Is it possible to view data from an IFC file with this library?

**Comment by jmirtsch** [comment]
Not from a user interface point of view.  This toolkit is for reading/writing files.  You can set a break point and see the relationships and attributes within you IDE, but I'm not sure that's what you are asking for.  We do build tools on top of this such as our IfcTreeViewer but it's not open source at this point in time.

**Reply by moaciroj** [reply]
OK, thank you,

So is it possible to read geometric and non-geometric information from a model in IFC?

Do you have any example code for reading an IFC file?

The examples that I had access to, created the data structures but didn't read pre-existing information.

I am writing a master's thesis and this information would be very important.

Regards,
Moacir

**Reply by jmirtsch** [reply]
There are some examples of reading (and writing) ifc files here, https://github.com/GeometryGym/GeometryGymIFCExamples

If you have problems and can share your code with explanations on data you need to query, I will do my best to help.
